src/lib/carplay/PVCarPlay.android.ts

```diff
@@ -1,55 +1,250 @@
+    if (currentTimeStamp - podcastRefreshDebounce > 1000) {
+      handleAndroidAutoPodcastsUpdate()
+      podcastRefreshDebounce = currentTimeStamp
+    }
+  } else if (mediaId.startsWith(MediaKeys.Podcast)) {
+    // mirrors handlePodcastsListOnSelect.
+    // load podcast if content needs to be refreshed (?), or content is empty.
+    // TODO: when is content needs to be refreshed? or always refresh?
+    // eslint-disable-next-line @typescript-eslint/tslint/config
+    if (browseTree[mediaId] === undefined) {
+      const index = mediaId.substring(MediaKeys.Podcast.length + 1)
+      const { subscribedPodcasts } = getGlobal()
+      const podcast = subscribedPodcasts[index] as Podcast
+      const queryEpisodes: Episode[] = (await getEpisodesForPodcast(podcast))[0] || []
+      cachedPodcasts[podcast.id] = podcast
+      cachedEpisodes[podcast.id] = queryEpisodes
+      updateAndroidAutoBrowseTree({
+        [mediaId]: queryEpisodes.map((episode, index) => {
+          const pubDate =
+            (episode?.liveItem?.start && readableDate(episode.liveItem.start)) ||
+            (episode.pubDate && readableDate(episode.pubDate)) ||
+            ''
+          return {
+            title: episode.title || translate('Untitled Episode'),
+            subtitle: pubDate,
+            playable: '0',
+            iconUri: episode.imageUrl || podcast.imageUrl || undefined,
+            mediaId: `${MediaKeys.Episode}-${podcast.id}-${index}`
+          }
+        })
+      })
+    }
+  }
+}
+
+export const handlePlayRemoteMediaId = async (mediaId: string) => {
+  if (mediaId.startsWith(MediaKeys.Queue)) {
+    const episodeId = mediaId.substring(MediaKeys.Queue.length + 1)
+    const foundEpisode = getQueue().filter((episode) => episode.episodeId === episodeId)
+    if (foundEpisode.length === 0) {
+      errorLogger(
+        _fileName,
+        'handlePlayRemoteMediaId',
+        `[Android Auto] ${mediaId} no longer exists in the current Queue.`
+      )
+      return
+    }
+    const convertedEpisode = convertNowPlayingItemToEpisode(foundEpisode[0])
+    loadEpisodeInPlayer(convertedEpisode, convertedEpisode.podcast)
+  } else if (mediaId.startsWith(MediaKeys.History)) {
+    const episodeId = mediaId.substring(MediaKeys.History.length + 1)
+    const foundEpisode = (await getHistory()).filter((episode) => episode.episodeId === episodeId)
+    if (foundEpisode.length === 0) {
+      errorLogger(
+        _fileName,
+        'handlePlayRemoteMediaId',
+        `[Android Auto] ${mediaId} no longer exists in the current History.`
+      )
+      return
+    }
+    const convertedEpisode = convertNowPlayingItemToEpisode(foundEpisode[0])
+    await loadEpisodeInPlayer(convertedEpisode, convertedEpisode.podcast)
+    refreshHistory()
+  } else if (mediaId.startsWith(MediaKeys.Episode)) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const [_, podcastId, index] = /^Episode-(.+)-(\d+)$/.exec(mediaId)!
```

Comment:
Do you mind explaining whats going on here a bit? How would the mediaId contain this kind of pattern in it?

Reply:
every "item" in the android auto navigator view only contains metadata and a unique mediaId; this mediaId is used to identify if an item is from episode, queue or history and is then processed accordingly. They are built in handleAndroidAutoPodcastsUpdate, handleAndroidAutoQueueUpdate and refreshHistory, where the mediaId is set for each item with the MediaKeys prefix